tests/e2e/support/page/files/allFiles.ts

```diff
@@ -111,6 +121,39 @@ export class AllFilesPage {
     return downloads
   }
 
+  async downloadFilesBatchAction({
+    names,
+    folder
+  }: {
+    names: string[]
+    folder: string
+  }): Promise<Download[]> {
+    const { page } = this.actor
+    const startUrl = page.url()
+    const downloads = []
+    if (folder) {
+      await filesCta.navigateToFolder({ page: page, path: folder })
+    }
+    for (const name of names) {
+      await this.toggleFileOrFolderCheckbox(page, true, name)
+    }
+    const [download] = await Promise.all([
+      page.waitForEvent('download'),
+      this.batchActionDownloadLocator.click()
+    ])
+    downloads.push(download)
+    await page.goto(startUrl)
+    return downloads
+  }
+
+  async toggleFileOrFolderCheckbox(page, enable, path) {
+    const checkBoxLocator = util.format(this.checkboxSelector, path)
+    if (enable) {
+      await page.check(checkBoxLocator)
+      expect(await page.isChecked(checkBoxLocator)).toBeTruthy()
```

Comment:
I think you don't need this check. As per the playwright doc `page.check()` asserts internally